.github/CONTRIBUTING.md

```diff
@@ -167,3 +167,71 @@ after the starting description.
  * ...
  */
 ```
+
+### Error Messages
+
+User-facing error messages should be clear, concise, and consistent. Error
+messages should be in sentence case but should not end with a period. Error
+messages should be free of grammatical errors and typos and written in American
+English.
+
+> ⚠️ Note that the error message style guide is a work in progress, and not all
+> the error messages have been updated to conform to the current styles. Error
+> message styles that should be followed:
+
+1. Messages should start with an upper case:
+
+```
+Bad: cannot parse input
+Good: Cannot parse input
+```
+
+2. Messages should not end with a period:
+
+```
+Bad: Cannot parse input.
+Good: Cannot parse input
+```
+
+3. Message should not use quotes for values:
+
+```
+Bad: Cannot parse input "x"
+Good: Cannot parse input x
+```
+
+4. Message should state the action that lead to the error:
+
+```
+Bad: Invalid input x
+Good: Cannot parse input x
+```
+
+5. Active voice should be used:
+
+```
+Bad: Input x cannot be parsed
+Good: Cannot parse input x
+```
+
+6. Messages should not use contractions:
+
+```
+Bad: Can't parse input x
+Good: Cannot parse input x
+```
+
+7. Messages should use a colon when providing additional information:
+
+```
+Bad: Cannot parse input x. value is empty
+Bad: Cannot parse input x: value is empty
+Good: Cannot parse input x, value is empty
```

Comment:
The example part seems having an error?
```suggestion
Good: Cannot parse input x: value is empty
```

Reply:
Opps. From the original suggestion it was "Should the message use a colon or comma when providing additional information". Do you prefer colon, comma, or either?

Reply:
We prefer colon.

Reply:
I've updated the guide to mention that colon is preferred.